pkg/pillar/zedcloud/authen.go

```diff
@@ -350,87 +350,82 @@ func VerifyCloudCertChain(ctx *ZedCloudContext, serverName string, content []byt
 		if cert.Type == zcert.ZCertType_CERT_TYPE_CONTROLLER_INTERMEDIATE {
 			ok := interm.AppendCertsFromPEM(cert.GetCert())
 			if !ok {
-				err := fmt.Errorf("VerifyCloudCertChain: fail to append intermediate certs")
+				err := fmt.Errorf("VerifySigningCertChain: fail to append intermediate certs")
 				return nil, err
 			}
-			log.Debugf("VerifyCloudCertChain: get intermediate cert %d, len %d\n",
+			log.Debugf("VerifySigningCertChain: get intermediate cert %d, len %d\n",
 				keyCnt, len(cert.GetCert()))
 		} else if cert.Type == zcert.ZCertType_CERT_TYPE_CONTROLLER_SIGNING {
 			if len(certByte) > 0 {
-				err := fmt.Errorf("VerifyCloudCertChain: received more than one leaf cert")
+				err := fmt.Errorf("VerifySigningCertChain: received more than one leaf cert")
 				return nil, err
 			}
 			certByte = cert.GetCert()
 			block, _ = pem.Decode(certByte)
 			if block == nil {
-				err := fmt.Errorf("VerifyCloudCertChain: can not get cert block")
+				err := fmt.Errorf("VerifySigningCertChain: can not get cert block")
 				return nil, err
 			}
 			leafcert, err = x509.ParseCertificate(block.Bytes)
 			if err != nil {
-				log.Errorf("VerifyCloudCertChain: fail to parse certificate %v\n", err)
+				log.Errorf("VerifySigningCertChain: fail to parse certificate %v\n", err)
 				return nil, err
 			}
-			log.Debugf("VerifyCloudCertChain: get signing cert %d, len %d\n",
+			log.Debugf("VerifySigningCertChain: get signing cert %d, len %d\n",
+				keyCnt, len(cert.GetCert()))
+		} else if cert.Type == zcert.ZCertType_CERT_TYPE_CONTROLLER_ECDH_EXCHANGE {
+			// skip for the CONTROLLER_ECDH_EXECHANGE cert type here, not for signing purpose
+			log.Debugf("VerifySigningCertChain: ECDH_EXCHANGE type, ignore cert %d, len %d\n",
 				keyCnt, len(cert.GetCert()))
 		} else {
-			err := fmt.Errorf("VerifyCloudCertChain: received cert with unknown type %d", cert.Type)
+			err := fmt.Errorf("VerifySigningCertChain: received cert with unknown type %d", cert.Type)
 			return nil, err
```

Comment:
I would suggest logging this and not returning an error. As we evolve the system there might be more cert types and there is no reason an old EVE should fail due to them. Can do that as a separate PR.